src/components/OrdersWidget/OrderRow.tsx

```diff
@@ -152,44 +87,147 @@ function calculatePrice(_numerator?: string | null, _denominator?: string | null
+      </div>
+    </div>
   )
+}
+
+interface UnfilledAmountProps extends Pick<Props, 'order' | 'pending'> {
+  sellToken: TokenDetails
+}
+
+const UnfilledAmount: React.FC<UnfilledAmountProps> = ({ sellToken, order, pending }) => {
   const unfilledAmount = useMemo(() => formatAmount(order.remainingAmount, sellToken.decimals) || '0', [
     order.remainingAmount,
     sellToken.decimals,
   ])
+  const unlimited = useMemo(() => order.priceDenominator.gt(MIN_UNLIMITED_SELL_ORDER), [order.priceDenominator])
+
+  return (
+    <div className={'container' + (unlimited ? '' : ' sub-columns two-columns')}>
+      {unlimited ? (
+        <Highlight color={pending ? 'grey' : ''}>no limit</Highlight>
+      ) : (
+        <>
+          <div>{unfilledAmount}</div>
+          <div>
+            <strong>{displayTokenSymbolOrLink(sellToken)}</strong>
+          </div>
+        </>
+      )}
+    </div>
+  )
+}
+
+interface AccountBalanceProps extends Pick<Props, 'order' | 'isOverBalance'> {
+  sellToken: TokenDetails
+}
+
+const AccountBalance: React.FC<AccountBalanceProps> = ({ sellToken, order, isOverBalance }) => {
   const accountBalance = useMemo(() => formatAmount(order.sellTokenBalance, sellToken.decimals) || '0', [
     order.sellTokenBalance,
     sellToken.decimals,
   ])
-  const unlimitedAmount = useMemo(() => order.priceDenominator.gt(MIN_UNLIMITED_SELL_ORDER), [order.priceDenominator])
 
+  return (
+    <div className="container sub-columns three-columns">
+      <div>{accountBalance}</div>
+      <strong>{displayTokenSymbolOrLink(sellToken)}</strong>
+      <div className="warning">{isOverBalance && <FontAwesomeIcon icon={faExclamationTriangle} />}</div>
+    </div>
+  )
+}
+
+const Expires: React.FC<Pick<Props, 'order' | 'pending'>> = ({ order, pending }) => {
   const { isNeverExpires, expiresOn } = useMemo(() => {
     const isNeverExpires = isBatchIdFarInTheFuture(order.validUntil)
     const expiresOn = isNeverExpires ? '' : formatDateFromBatchId(order.validUntil)
+
     return { isNeverExpires, expiresOn }
   }, [order.validUntil])
 
+  return <div>{isNeverExpires ? <Highlight color={pending ? 'grey' : ''}>Never</Highlight> : expiresOn}</div>
+}
+
+async function fetchToken(
+  tokenId: number,
+  networkId: number,
+  setFn: React.Dispatch<React.SetStateAction<TokenDetails | null>>,
+): Promise<void> {
+  const token = await getTokenFromExchangeById({ tokenId, networkId })
+
+  // It is unlikely the token ID coming form the order won't exist
+  // Still, if that ever happens, store null and keep this order hidden
+  setFn(token)
+}
+
+interface Props {
+  order: AuctionElement
+  isOverBalance: boolean
+  networkId: number
+  pending?: boolean
+}
+
+const OrderRow: React.FC<Props> = props => {
+  const { order, networkId, pending = false } = props
+
+  // Fetching buy and sell tokens
+  const [sellToken, setSellToken] = useState<TokenDetails | null>(null)
+  const [buyToken, setBuyToken] = useState<TokenDetails | null>(null)
```

Comment:
`useSafeState`?

Reply:
For some reason it doesn't work with `useSafeState`. With `useState` it does.

The ui does not update async when the token is fetched.

Reply:
Well, now it did work. Go figure...